rpc/rpc-transport/socket/src/socket.c

```diff
@@ -939,33 +938,18 @@ __socket_server_bind(rpc_transport_t *this)
             ((struct sockaddr_in *)&this->myinfo.sockaddr)->sin_port = htons(
                 sin_port);
         }
-        retries = 10;
-        while (retries) {
-            ret = bind(priv->sock, (struct sockaddr *)&this->myinfo.sockaddr,
-                       this->myinfo.sockaddr_len);
-            if (ret != 0) {
-                gf_log(this->name, GF_LOG_ERROR, "binding to %s failed: %s",
-                       this->myinfo.identifier, strerror(errno));
-                if (errno == EADDRINUSE) {
-                    gf_log(this->name, GF_LOG_ERROR, "Port is already in use");
-                    ret = -EADDRINUSE;
-
-                    /* TODO: Remove this delay. It was added only to address
-                     * failures in the regression test suite, In a real
-                     * situation, if a port is in use when the process starts,
-                     * it most likely will remain in use during 10 seconds as
-                     * well.*/
-                    sleep(1);
-                    retries--;
-                } else {
-                    break;
-                }
-            } else {
-                break;
+
+        ret = bind(priv->sock, (struct sockaddr *)&this->myinfo.sockaddr,
+                   this->myinfo.sockaddr_len);
+        if (ret != 0) {
+            gf_log(this->name, GF_LOG_ERROR, "binding to %s failed: %s",
+                   this->myinfo.identifier, strerror(errno));
+            if (errno == EADDRINUSE) {
```

Comment:
nit: 
1. errno might be changed by gf_log()
2. Why would you log 2 errors? Why not if (errno == EAADRINUSE) logA else logB

Reply:
Yes...actually, the other log is not needed because the previous log function would log the same message i.e, `Port already in use` if we get the errno as `EADDRINUSE`. I will remove it.